ephios/extra/templates/extra/widgets/recurrence_picker.html

```diff
@@ -0,0 +1,231 @@
+                                   v-model="rule.until" @focus="rule.end_mode='UNTIL'"
+                                   :required="rule.end_mode=='UNTIL'">
+                        </div>
+                        <div v-if="pickHour" class="col-4">
+                            <input type="time" class="form-control form-control-sm"
+                                   v-model="rule.UNTIL_TIME" @focus="rule.end_mode='UNTIL'">
+                        </div>
+
+                    </div>
+
+                    <div class="row">
+                        <div class="col-1">
+                            <input class="form-check-input" type="radio" name="end_mode" v-model="rule.end_mode"
+                                   value="COUNT" required>
+                        </div>
+                        <span class="col-2">
+                            {% translate "for" %}
+                        </span>
+                        <div class="col-5 col-md-7">
+                            <input type="text" class="form-control form-control-sm flex-fill"
+                                   v-model="rule.count" @focus="rule.end_mode='COUNT'"
+                                   :required="rule.end_mode=='COUNT'">
+                        </div>
+                        <span class="col-4 col-md-2">
+                            {% translate "ocurrences" %}
+                        </span>
+                    </div>
+                </div>
+            </li>
+            <li class="list-group-item p-3" v-for="date in dates"
+                :class="{'border-danger': !(date.date && (!pickHour || date.time))}">
+                <div class="row align-items-center">
+                    <div class="col-5">
+                        <input type="date" v-model="date.date" class="form-control" required>
+                    </div>
+                    <div class="col-5">
+                        <input type="time" v-if="pickHour" v-model="date.time" class="form-control" required>
+                    </div>
+                    <div class="col-1 ms-auto me-3">
+                        <button type="button" class="btn btn-danger" @click="removeDate(date)">
+                            <i class="fas fa-trash"></i><span
+                                class="visually-hidden">{% translate "Remove" %}</span>
+                        </button>
+                    </div>
+                </div>
+            </li>
+        </ul>
+        <div class="form-group">
+            <button class="btn btn-secondary ms-1" id="add-rule" @click.stop.prevent="addRule()"><span
+                class="fa fa-plus"></span> Add rule
+            </button>
+            <button class="btn btn-secondary ms-1" id="add-date" @click.stop.prevent="addDate()"><span
+                class="fa fa-plus"></span> Add date
+            </button>
+            <button type="submit" class="btn btn-primary float-end"><span
+                class="fa fa-copy"></span> {% translate "Copy" %}</button>
```

Comment:
The submit button should not live in the widget code, but in the templates using it. Also both are missing an abort button to go back to event detail page.

Reply:
ahhh but that makes it rather ugly because the copy button cannot be placed where it currently is

Reply:
I'm sure you can make it work 🤔
The date list could be scrollable so content below the widget doesn't get flushed to the abyss.

Reply:
well as the button is not below the date list this is not the issue here
![image](https://github.com/user-attachments/assets/dd78091f-0f7d-4009-a89d-ea45bc6cc818)

Reply:
but i like where the confirm button is :P

but srsly, do you have an idea for how to layout this with the confirm button below so it does not look bad?

Reply:
I think on the very right would make sense, where we have the save button in other forms as well. It makes sense looking over the date list to confirm they are correct, then hit save.

Reply:
hm that only works well for a short list of dates, I'm not really a fan of having a vertically scrollable container there. but maybe thats fine because people probably rarely have more than 20 occurrences?

Reply:
I moved the buttons out, please check if you like the placement